packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -82,7 +82,49 @@ The POC is found in this branch [Worker Rendering POC](https://github.com/ni/nim
 
 ### Data Structure and Interface
 
-We have two possible solutions for representing the data in the memory. They will be decided with a spec update. The fist one is an in-house solution:
+The best solution to solve the API of the wafermap is to use parts of both of the proposed methods mentioned in the next section. This means using Apache Arrow as the wafer component API and Typed Arrays for their iterating performance and transferability to worker threads.
+
+The Public API will be the following:
+
+```TS
+public diesTable: Table<{
+        colIndex: Int32,
+        rowIndex: Int32,
+        value: Float32,
+        tags: Uint32;
+        metadata: never;
+    }>
+```
+
+This will be the [Apache Arrow](https://arrow.apache.org/docs/js/classes/Arrow_dom.Table.html) table schema.
+The row and column indices will be `Int32` columns, the values will be `Float32` columns.
+The tags for each die will be represented as a 32 bit mask stored in a `Uint32` column.
+The metadata column will be stored in an wildcard typed column.
+
+This approach has the benefits of a row based format that aligns well with the existing public API, as well as a nice public API that easily allows future improvements.
+
+The limits for this approach are the following:
+
+1. There seems to be no support for columns of lists of strings. We decided to overcome this using a bit mask of tags. Another possible solution can be a dynamic number of columns for storing tags, but the performance may suffer.
+2. There is no support currently for [searching or filtering the table](https://github.com/apache/arrow/issues/13233). The possible solutions for this are searching by iterating over the whole table, which is not feasible (see 4.) or using a higher level library such as [aquero](https://uwdata.github.io/arquero/).The solution we chose is using a custom method for finding rows based on column and row indexes cached as typed arrays. This method provides faster access to row values and metadata and does not induce additional dependencies.
```

Comment:
Can you clarify what wafer map features require searching or filtering the table? Is it something that happens on every data update or just when certain attributes change?